src/execution_plan/ops/op_update.c

```diff
@@ -96,31 +98,34 @@ static void _UpdateEdge(OpUpdate *op, EntityUpdateCtx *ctx) {
 	SIValue *old_value = GraphEntity_GetProperty((GraphEntity *)edge, ctx->attr_id);
 
 	if(old_value == PROPERTY_NOTFOUND) {
-		// Add new property.
+		// Adding a new property; do nothing if its value is NULL.
+		if(SI_TYPE(ctx->new_value) == T_NULL) return false;
 		GraphEntity_AddProperty((GraphEntity *)edge, ctx->attr_id, ctx->new_value);
 	} else {
 		// Update property.
 		GraphEntity_SetProperty((GraphEntity *)edge, ctx->attr_id, ctx->new_value);
 	}
+	return true;
 }
 
 /* Executes delayed updates. */
 static void _CommitUpdates(OpUpdate *op) {
+	uint properties_set = 0;
 	for(uint i = 0; i < op->pending_updates_count; i++) {
 		EntityUpdateCtx *ctx = &op->pending_updates[i];
 		// Map the attribute key if it has not been encountered before
 		if(ctx->attr_id == ATTRIBUTE_NOTFOUND) {
 			ctx->attr_id = GraphContext_FindOrAddAttribute(op->gc, ctx->attribute);
 		}
 		if(ctx->entity_type == GETYPE_NODE) {
-			_UpdateNode(op, ctx);
+			properties_set += _UpdateNode(op, ctx);
```

Comment:
`_UpdateNode` returns `bool`, I find it strange to be adding `bool` to `uint`
either update `_UpdateNode` and `_UpdateEdge` return type or have an `if` condition here

Reply:
:+1: Switched return types to ints.